### PR TITLE
Produce KoreBuild as a compiled artifact

### PR DESCRIPTION
**Pull request by natemcmaster**
- Refactored KoreBuild scripts to be reusable functions
- Removed KoreBuild's dependency on Microsoft.Common.targets
- Fixed a minor bug in ZipArchive task
- Add functional tests for a simple KoreBuild repo
- Rewrite the bootstrapper scripts to pull from Azure storage blobs

Part of https://github.com/aspnet/KoreBuild/issues/258

### CODE REVIEW

**Comment by natemcmaster** [comment]
See https://github.com/aspnet/specs/blob/master/runtime/design-notes/2017-06-19-build-korebuild.md for the high-level description of what this is about.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+language: csharp
```
Comment (natemcmaster) [inline .travis.yml]: @pranavkm do you have the ability to enable Travis for this repo? It would be good to enable so we can ensure the functional tests pass on Linux

Comment (pranavkm) [inline .travis.yml]: I don't think so. Perhaps @muratg might?

Comment (natemcmaster) [inline .travis.yml]: Hmm, @muratg doesn't either, and @Eilon is OOF.

File: test/KoreBuild.FunctionalTests/KoreBuild.FunctionalTests.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+<Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
```
Comment (pranavkm) [inline test/KoreBuild.FunctionalTests/KoreBuild.FunctionalTests.csproj]: I'm wary of doing this - if we were to bootstrap from scratch, this ends up being a cycle. Can we just copy this file from source \ use the one that's already here?

File: test/BuildTools.Tasks.Tests/ZipArchiveTest.cs [hunk 3]
```diff
@@ -92,6 +92,40 @@ public void CreatesZip()
             }
         }
 
+        [Fact]
+        public void FailsForEmptyFileName()
```
Comment (pranavkm) [inline test/BuildTools.Tasks.Tests/ZipArchiveTest.cs]: Separate commit. Seems entirely unrelated to the KoreBuild change.

File: sdk/KoreBuild/modules/sharedsources/sharedsources.csproj [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <Pack>true</Pack>
+      <PackagePath>$(ContentTargetFolders)\any\any\</PackagePath>
+    </EmbeddedResource>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Update="@(PackageReference)" PrivateAssets="All" />
+  </ItemGroup>
+
+</Project>
```
Comment (pranavkm) [inline sdk/KoreBuild/modules/sharedsources/sharedsources.csproj]: new lines at EOF

File: test/KoreBuild.FunctionalTests/KoreBuild.FunctionalTests.csproj (hunk 2 above)
Comment (natemcmaster) [inline test/KoreBuild.FunctionalTests/KoreBuild.FunctionalTests.csproj]: That's a good observation, and one I plan to address in a follow-up as a part of the "version pinning" work. For now, this repo already has a bunch of dependencies on aspnetcore packages, so it will be a big change that will affect existing tools.

File: test/BuildTools.Tasks.Tests/ZipArchiveTest.cs (hunk 3 above)
Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/ZipArchiveTest.cs]: I discovered this bug it while using the ZipArchive task to produce the korebuild artifact.

Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/ZipArchiveTest.cs]: but I'll move to separate commit to keep history cleaner
